pkg/pool-weighted/contracts/CustomFeeAuthorizer.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+
+contract CustomFeeAuthorizer {
+    event feeSetterAdded(address indexed feeSetter);
+    event feeSetterRemoved(address indexed feeSetter);
+
+    mapping(address => bool) private isCustomFeeSetter;
+    bool public isCustomFeeEnabled = false ;
+    address public solver;
+
+    function canSetCustomFee(address _setterAddress) public view returns(bool _isAuth){
+        if(isCustomFeeEnabled){
+            _isAuth = (isCustomFeeSetter[_setterAddress] ||  solver == _setterAddress) ;
+        }else{
+        _isAuth = false;
+        }
+    }
+
+    function addCustomFeeSetter(address _toAdd) onlySolver public {
+        require(_toAdd != address(0));
+        require(isCustomFeeEnabled,"Custom Fee Not Enabled");
+        isCustomFeeSetter[_toAdd] = true;
+        emit feeSetterAdded(_toAdd);
+    }
+
+    function removeCustomFeeSetter(address _toRemove) onlySolver public {
+        require(_toRemove != msg.sender);
+        isCustomFeeSetter[_toRemove] = false;
+        emit feeSetterRemoved(_toRemove);
+    }
+
+    function enableCustomFee() onlySolver() internal {
+        require(!isCustomFeeEnabled, "Already Enabled");
+        isCustomFeeEnabled = true;
+    }
+
+    function _setSolverAddress(address _solver) internal {
```

Comment:
Shouldn't this be an external function with the onlySolver() modifier?

Reply:
if we can mark `isCustomFeeEnabled` as immutable then we can  call this function in the constructor based on the value of `isCustomFeeEnabled`.
 But if not then initially 0 address will be marked as solver so we cannot use `onlySolver` here. But we can have a logic where initially owner() can call this function and after that `onlySolver` can call that

Reply:
> @markusbkoch who should set the solver? Governance, or is it fixed at pool creation time? Can the existing solver pass the torch to another address?

Governance should not be able to set it. Fixed would be good enough, solver being able to pass it on would be a nice to have